Parsers/ASimWebSession/Parsers/ASimWebSessionCiscoMeraki.yaml

```diff
@@ -0,0 +1,228 @@
+Parser:
+  Title: Web Session ASIM parser for Cisco Meraki
+  Version: '0.1'
+  LastUpdated: Jun 20 2023
+Product:
+  Name: Cisco Meraki
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Cisco Meraki Documentation
+  Link: https://documentation.meraki.com/
+Description: |
+  This ASIM parser supports normalizing Cisco Meraki logs to the ASIM Web Session normalized schema.
+ParserName: ASimWebSessionCiscoMeraki
+EquivalentBuiltInParser: _ASim_WebSession_CiscoMeraki
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let ActionLookup = datatable (action: string, DvcAction: string)  [
+    'allow', 'Allow',
+    'log', 'Allow',
+    'accept', 'Allow',
+    'block', 'Deny',
+    'deny', 'Deny',
+    'quarantine', 'Deny'
+  ];
+  let parser=(disabled: bool=false) {
+      union isfuzzy=true
+          (
+          meraki_CL
+          | project-rename LogMessage =  Message
+          ),
+          (
+          Syslog
+          | project-rename LogMessage =  SyslogMessage
+          )
+      | where not(disabled) and (LogMessage has "urls" or LogMessage has_all("security_event", "security_filtering_file_scanned"))
+      | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
+      | extend
+          LogType = tostring(Parser[2]),
+          Substring = tostring(Parser[3])
+      | extend LogSubType = iff(LogType == "security_event", tostring(extract(@"^(\S+)\s", 1, Substring)), "")
+      | where (LogType == "urls" or (LogType == "security_event" and LogSubType == "security_filtering_file_scanned"))
+      | parse-kv Substring as (src: string, dst: string, url: string, disposition: string, action: string, sha256: string, name: string, mac: string) with (pair_delimiter=" ", kv_delimiter="=")
+      | parse kind=regex Substring with * "request:[ ]{0,1}" request " "urls
+      | parse Substring with * " sha256" fsha256 " "restmessage
+      | parse Substring with * " agent=" agent " request:" restmessage
+      | extend src= iff(
+          src has '"',
+          replace_string(tostring(src), '"', ''),
+          replace_string(tostring(src), "'", "")
+      ),
+          dst= iff(
+          dst has '"',
+          replace_string(tostring(dst), '"', ''),
+          replace_string(tostring(dst), "'", "")
+      )
+      | parse src with *  "[" temp_srcip "]:" temp_srcport
+      | parse dst with *  "[" temp_dstip "]:" temp_dstport
+      | extend
+          Epoch = tostring(Parser[0]),
+          Device = tostring(Parser[1])
+      | extend
+          EpochTimestamp = split(Epoch, ".")
+      | extend
+          EventStartTime = unixtime_seconds_todatetime(tolong(EpochTimestamp[0]))
```

Comment:
can we do above two steps for extracting and converting startime in one line?

Reply:
Done